node/remote/utils.go

```diff
@@ -16,12 +17,13 @@ var (
 	HTTPProtocols = regexp.MustCompile("https?://")
 )
 
-// GetHeightRequestHeader returns the grpc.CallOption to query the state at a given height
-func GetHeightRequestHeader(height int64) grpc.CallOption {
-	header := metadata.New(map[string]string{
-		grpctypes.GRPCBlockHeightHeader: strconv.FormatInt(height, 10),
-	})
-	return grpc.Header(&header)
+// GetHeightRequestContext adds the height to the context for querying the state at a given height
+func GetHeightRequestContext(context context.Context, height int64) context.Context {
```

Comment:
Does this work the exact same way of the request header? Would this work with Cosmos gRPC properly?

Reply:
the request header did not work. I used the way introduced in the [cosmos sdk tutorial](https://docs.cosmos.network/master/run-node/interact-node.html#query-for-historical-state-using-go) which includes the height request in the call context and turns out that actually worked.

We will have to update the bdjuno code if this PR is merged.

Reply:
All right thanks for the clarification